src/control/server/storage/config.go

```diff
@@ -142,10 +149,52 @@ type BdevConfig struct {
 	Hostname    string    `yaml:"-"` // used when generating templates
 }
 
+func (bc *BdevConfig) checkNonZeroFileSize() error {
+	if bc.FileSize == 0 {
+		return errors.Errorf("bdev_class %s requires non-zero bdev_size",
+			bc.Class)
+	}
+
+	return nil
+}
+
+// Validate sanity checks engine bdev config parameters.
 func (bc *BdevConfig) Validate() error {
+	if common.StringSliceHasDuplicates(bc.DeviceList) {
+		return errors.New("bdev_list contains duplicate pci addresses")
+	}
```

Comment:
Doesn't necessarily need to be fixed for this PR, but it looks like `common.StringSliceHasDuplicates()` is case-sensitive, meaning that there is still a chance that a device could be specified more than once with different cases.

Reply:
good catch